website/source/docs/guides/consul-splitting.md

```diff
@@ -0,0 +1,461 @@
+contains some changes you want to ship in a canary deployment.
+
+![Architecture diagram of the splitting demo. A web service directly connects to two different versions of the API service through proxies. Consul configures those proxies.](/static/img/consul-splitting-architecture.png)
+
+## Start the Environment
+
+First clone the repo containing the source and examples for this guide post.
+
+```shell
+$ git clone git@github.com:hashicorp/consul-demo-traffic-splitting.git
+```
+
+Change directories into the cloned folder, and start the demo environment with
+`docker-compose up`. This command will run in the foreground, so you’ll need to
+open a new terminal window after you run it.
+
+```shell
+$ docker-compose up
+
+Creating consul-demo-traffic-splitting_api_v1_1    ... done
+Creating consul-demo-traffic-splitting_consul_1 ... done
+Creating consul-demo-traffic-splitting_web_1    ... done
+Creating consul-demo-traffic-splitting_web_envoy_1    ... done
+Creating consul-demo-traffic-splitting_api_proxy_v1_1 ... done
+Attaching to consul-demo-traffic-splitting_consul_1, consul-demo-traffic-splitting_web_1, consul-demo-traffic-splitting_api_v1_1, consul-demo-traffic-splitting_web_envoy_1, consul-demo-traffic-splitting_api_proxy_v1_1
+```
+
+The following services will automatically start in your local Docker environment
+and register with Consul.
+
+- Consul Server
+- Web service with Envoy sidecar
+- API service version 1 with Envoy sidecar
+
+Consul is preconfigured to run as a single server, with with all the
+configurations for splitting enabled:
+
+- Connect is enabled - Traffic shaping requires that you use Consul Connect.
+
+- gRPC is enabled - splitting also requires the you use Envoy as a sidecar
+proxy, and Envoy gets its configuration from Consul via gRPC.
+
+- Central service configuration is enabled - you will use configuration entries
+to specify that the API service protocol, and define your splitting ratios.
+
+The Consul configuration file at `consul_config/consul.hcl` contains the
+follwoing.
+
+```hcl
+data_dir = "/tmp/"
+log_level = "DEBUG"
+server = true
+
+bootstrap_expect = 1
+ui = true
+
+bind_addr = "0.0.0.0"
+client_addr = "0.0.0.0"
+
+connect {
+  enabled = true
+}
+
+ports {
+  grpc = 8502
+}
+
+enable_central_service_config = true
+```
+
+You can find the service definitions for this demo in the `service_config`
+folder.
+
+Once everything is up and running, you can view the health of the registered
+services by checking the Consul UI at `http://localhost:8500`. All services
+should be passing their health checks.
+
+![List of services in the Consul UI including Consul, and the web and API services with their proxies](/static/img/consul-splitting-services.png)
+
+Curl the Web endpoint to make sure that the whole application is running. The
+Web service will get a response from version 1 of the API service.
+
+```hcl
+$ curl localhost:9090
+Hello World
+###Upstream Data: localhost:9091###
+  Service V1%
```

Comment:
Haha, I thought this too and then just glossed over it as I got more used to it. @nicholasjackson any insights?

Reply:
I was wondering this too and just got used to it and forgot! @nicholasjackson any insight?

Reply:
Nic says that this has something to do with the command output because we're using curl. Sounds like it would be annoying to hide, but we can explain it briefly if you think it's worth it. My inclination is to leave it as is, since I don't think it's going to change people's perception of Consul functionality.

Reply:
@Paul I think your last comment is displaying strangely. Copying it back here for posterity: 

> Hmm, I don't think it's huge but I do feel like it confused me especially
given that we are talking about splitting percentages. I thought at first
it was trying to be smart and display the current split percentage but
realised that was strange. Does it go away with `curl -q`?

🤔 It doesn't go away. I'll google around more and see what I can do

Reply:
It seems like the `%` is something zsh adds because there is no new line in the output that the API service returns. It looks even more confusing if you use bash; it appends the next command prompt on to the end of the output from the service: `  Service V1bash-3.2$`.

I think the best fix is to add a newline to the service output. I submitted an issue, and If I have time I can dig around more and try to fix it myself. https://github.com/hashicorp/consul-demo-traffic-splitting/issues/4

Reply:
Oooh good detection skillz.

In that case I'd be tempted just to remove it from the guide and publish. We can add a new line to the code maybe but I think it's less confusing if people see their own shell behaviour than wonder what this is reading the guide.

If you want a solution that looks completely authentic you could add `-w "\m"` to the curl command which [will add the newline for you](https://stackoverflow.com/questions/12849584/automatically-add-newline-at-end-of-curl-response-body). But I don't personally think that's necessary for our users to understand the guide or even follow along.